### ISSUE TITLE
Rewrite gui to be more modern looking

### ISSUE THREAD

**Issue by michel-cf** [parallaxinc/BlocklyPropClient#20]
Options:
- Kivy
- QT (PySide)
- GTK

**Comment by DarkmatterVale** [comment]
This sounds like a great idea and is really needed. QT will be a great substitute. PropellerIDE looks great and is written using a QT GUI. Do you have a preference?

**Comment by michel-cf** [comment]
I've been looking at GTK (actually already did a new branch, but it didn't
want to run on my pc)

Kivy looks the nicest, and I'm currently working on a QT version test.

**Comment by DarkmatterVale** [comment]
Yes, I took a look at Kivy and I agree that it is very sleek looking.

Should I merge my PR before we do this, or would you like me to wait for the GUI change?

**Comment by michel-cf** [comment]
You can merge it if you want, so I can see what needs to be implemented. But if you're not ready to merge, a feature list should suffice.

**Comment by DarkmatterVale** [comment]
I am almost ready to merge; I just need to know how to implement the includes part of the library finder. Would you like me to merge without that? It is not required since the program will default to using the included libraries.

Everything else is done though, including file saving and a box that contains the most recently compiled code (not editable yet).

**Comment by michel-cf** [comment]
Sure, you can merge it. The library finder and such will have to change anyway.

**Comment by michel-cf** [comment]
Although it might not look the fancy, wxPython provides a native looking interface.

Once all test implementations are done, a decision can be made

**Comment by DarkmatterVale** [comment]
Yes, they will have to be changed to work with the include part of each library. Is that what you are referring to?

I will merge right now.

**Comment by michel-cf** [comment]
Once a decision is made on what library to use based on my tests (in separate branches) the remaining functionality will need to be ported.

It is not so that I want everyone to now install QT, wxPython, GTK, Kivy,... I'll be making executable builds that don't require installation once I find the test version sufficient and distribute these.

After a choice of technology is made, those who want to develop will have to install the chosen gui library.

**Comment by michel-cf** [comment]
A wxPython test version is now available for download on: http://owncloud.creatingfuture.eu/public.php?service=files&t=ab818b8ce686467b6afc72b0e3fba11f

I've been unable to get the GTK version to run,

**Comment by michel-cf** [comment]
Kivy version, not functional yet but the graphical components are there.

![blocklypropclient kivy](https://cloud.githubusercontent.com/assets/5745991/7631602/ab1ab9ce-fa44-11e4-898b-74ae2d27c5c8.png)

**Comment by DarkmatterVale** [comment]
That looks great! Very sleek and modern looking. Do you think this one looks the best so far?

**Comment by michel-cf** [comment]
I don't know. Although it might look nice, But the layout systems are not as good as the others, and if you resize the window, the components inside only resize when you release it. It doesn't resize while you hold it.

Defining this layout has been done using a text file (not in code), so I still need to figure out how to handle the events from within the code.

**Comment by DarkmatterVale** [comment]
@michel-cf What is the status of the client? Are you still working with Python in the new client?

**Comment by michel-cf** [comment]
Yes, I'm still working with Python (although I've been experimenting with C# to create a windows only version. I'm hoping I can make that one start automaticaly with Windows and minimize to the system tray/notification area)

I have not been working on it using new technologies. Because of the new project setup (cloud-compiling), the requirements for the application have changed a bit.
